### PR TITLE
creates script to send API calls hugging face models

### LINKED ISSUE #1: Research how to use different models from hugging face to perform question answering task
## Description 

The app needs to use AI models to power the interactions. Since we will be unable to host any models we should utilise prehosted models. One option is HuggingFace or OpenAI.

## Coversations to resolve
- [x] Which models are available on HuggingFace as hosted and ready for inference?
- [x] Which platform would be cheaper?

## Acceptance criteria
- [ ] Model provider is selected
- [ ] Prototype small script demonstrating communication with the model provider to prompt a model

### PR DESCRIPTION
**Pull request by aaroosh-07**
this PR creates a python script to communicate with hugging face models via post request to perform question answering task on a given context.

### CODE REVIEW

**Comment by prattyushmangal** [comment]
You need to add a readme first of all describing what this repo is and how you can run this script. 

Also is this marking the decision to go with HF instead of OpenAI? This pr or the ticket should address the conversations to resolve in the issue with a ADR (Architecture Decision Record markdown) too.

**Comment by prattyushmangal** [comment]
You also need to provide a venv and packages guide via a requirements.txt to enable people to recreate your environment and run your script.

**Comment by aaroosh-07** [comment]
I have added requirements.txt file to show all the modules needed to run the script and also updated readme file to include a small description of the project and steps to install all dependencies and run the script

**Comment by aaroosh-07** [comment]
Yes we are going ahead with huggingFace models as OpenAI provide with limited free credits and after that we will need to purchase the credits. In developement phase we can use hugging face models to build the application and then add support for OpenAI separately

File: README.md [hunk 1]
```diff
@@ -1 +1,47 @@
...
+
+### Setting up Environment
+
+1. Create python virtual environment using this command
+
+```
+python -m venv virtualenv
+```
+
+2. Install all the required modules.
```
Comment (prattyushmangal) [inline README.md]: You need to activate the virtual environment before you can install the requirements. If you install the requirements before you start you env then the installations will take place in your global python installation which is definitely not what you want to do.

**Review by prattyushmangal — CHANGES_REQUESTED** [review]
The main script is fine, but we should also add a script which uses the HF SDK instead of directly using the REST API from HF. Using the SDK will give us access to greater functionality and exception handling.

**Comment by prattyushmangal** [comment]
Also lets put these experimental scripts in an `experiments` directory or something.

**Comment by aaroosh-07** [comment]
> The main script is fine, but we should also add a script which uses the HF SDK instead of directly using the REST API from HF. Using the SDK will give us access to greater functionality and exception handling.

There is only transformers SDK provided by HF, which allows to download models and run them locally, but no SDK is present to use models deployed on serverless infrastructure by HF and we can only prompt them by sending REST API calls.

File: README.md (hunk 1 above)
Comment (aaroosh-07) [reply README.md]: I will add a step to activate virtual environment before installing requirements.

File: README.md [hunk 2]
```diff
@@ -1 +1,57 @@
...
+2. create a `.env` file in root directory of your project.
+3. add the following content in your `.env` file
+
+```
+API_TOKEN = "<your api token>"
+```
+
+### Running the script
+
+1. Activate the virtual environment
```
Comment (prattyushmangal) [inline README.md:43]: Why is this here if this is already in the Setting up the environment section?

**Review by prattyushmangal — APPROVED** [review]
Little things - pay closer attention to detail, but tackle this in the next pull request.
